scaleway/data_source_baremetal_offer_beta.go

```diff
@@ -0,0 +1,205 @@
+							Computed:    true,
+							Description: "Number of threads",
+						},
+					},
+				},
+			},
+			"disk": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Disk specifications of the offer",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Type of disk",
+						},
+						"capacity": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Capacity of the disk in byte",
+						},
+					},
+				},
+			},
+			"memory": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Memory specifications of the offer",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Type of memory",
+						},
+						"capacity": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Capacity of the memory in byte",
+						},
+						"frequency": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Frequency of the memory",
+						},
+						"ecc": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "True if error-correcting code is available on this memory",
+						},
+					},
+				},
+			},
+			"price_per_sixty_minutes": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Price of the offer for the next 60 minutes (a server order at 11h32 will be payed until 12h32)",
+			},
+			"price_per_month": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Price of the offer per months",
+			},
+			"quota_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Quota name of this offer",
+			},
+			"stock": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Stock status for this offer",
+			},
+		},
+	}
+}
+
+func dataSourceScalewayBaremetalOfferBetaRead(d *schema.ResourceData, m interface{}) error {
+	meta := m.(*Meta)
+	baremetalApi, fallBackZone, err := getBaremetalAPIWithZone(d, meta)
+	if err != nil {
+		return err
+	}
+
+	zone, offerID, _ := parseZonedID(datasourceNewZonedID(d.Get("offer_id"), fallBackZone))
+	res, err := baremetalApi.ListOffers(&baremetal.ListOffersRequest{
+		Zone: zone,
+	}, scw.WithAllPages())
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(res.Offers); i++ {
+		offer := res.Offers[i]
+		switch {
+		case offer.Name == d.Get("name"), offer.ID == offerID:
+			if !offer.Enable && !d.Get("allow_disabled").(bool) {
+				return fmt.Errorf("offer %s (%s) found in zone %s but is disabled. Add allow_disabled=true in your terraform config to use it.", offer.Name, offer.Name, zone)
+			}
+		default:
+			res.Offers = append(res.Offers[:i], res.Offers[i+1:]...)
+			i--
```

Comment:
I kwnow this code works but I found it hard to read 🤔

Reply:
I can, append the matching offers in a new slice, will be much more clear